astropy/table/serialize.py

```diff
@@ -51,52 +56,44 @@ def _represent_mixin_as_column(col, name, new_cols, mixin_cols,
     # - description: DO store
     # - meta: DO store
     info = {}
-    for attr, nontrivial, xform in (('unit', lambda x: x not in (None, ''), str),
+    for attr, nontrivial, xform in (('unit', lambda x: x is not None and x != '', str),
```

Comment:
Work around bug in unit where doing an equality test of `None` with an invalid unit raises an exception.

Reply:
Please add a comment in the code for this

Reply:
Better still, raise an issue!  That should be relatively trivial to fix.

Reply:
New issue #7603.

I'm not adding a comment about this because in retrospect the `x not in (None, '')` code is non-ideal anyway. It effectively expands to testing `x == None` instead of the preferred `x is None` test.